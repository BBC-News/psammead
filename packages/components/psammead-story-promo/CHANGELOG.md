# Psammead Media Indicator Changelog

<!-- prettier-ignore -->
| Version | Description |
| ------- | ----------- |
| 1.0.0   | [PR#507](https://github.com/bbc/psammead/pull/507) Bump to major version, no other changes. |
| 0.1.2   | [PR#488](https://github.com/BBC-News/psammead/pull/488) Hide story summary on device width lower than 600px. |
| 0.1.1   | [PR#474](https://github.com/BBC-News/psammead/pull/474) Update Story promo headline to use Great Primer. |
| 0.1.0   | [PR#453](https://github.com/BBC-News/psammead/pull/453) Create initial package. |
