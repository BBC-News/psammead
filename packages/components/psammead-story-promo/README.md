# ⛔️ This is an alpha component ⛔️

This component is currently tagged as alpha and is not suitable for production use. Following the passing of an accessibility review this component will be marked as ready for production and the alpha tag removed.

# psammead-story-promo - [![Known Vulnerabilities](https://snyk.io/test/github/bbc/psammead/badge.svg?targetFile=packages%2Fcomponents%2Fpsammead-story-promo%2Fpackage.json)](https://snyk.io/test/github/bbc/psammead?targetFile=packages%2Fcomponents%2Fpsammead-story-promo%2Fpackage.json) [![Storybook](https://raw.githubusercontent.com/storybooks/story-promo/master/badge/badge-storybook.svg?sanitize=true)](https://bbc.github.io/psammead/?path=/story/story-promo--default) [![GitHub license](https://img.shields.io/badge/license-Apache%202.0-blue.svg)](https://github.com/bbc/psammead/blob/latest/LICENSE) [![npm version](https://img.shields.io/npm/v/@bbc/psammead-story-promo.svg)](https://www.npmjs.com/package/@bbc/psammead-story-promo) [![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg)](https://github.com/bbc/psammead/blob/latest/CONTRIBUTING.md)

## Description

The `StoryPromo` component is designed to be used on 'index' pages, which are pages that link to other articles/stories. It supports having an image on the left of the promo with info on the right. This info can be any collection of nodes, however typically these would be a headline, text summary and timestamp.

## Installation

`npm install @bbc/psammead-story-promo`

## StoryPromo Props

<!-- prettier-ignore -->
| Argument | Type   | Required | Default | Example        |
| -------- | ------ | -------- | ------- | -------------- |
| image    | node   | No       | Null    | `<img>`          |
| info     | node   | Yes      | N/A     | `<h2>Title</h2>` |
| mediaIndicator     | node   | No      | null     | `<MediaIndicator duration="2:15" datetime="PT2M15S" offscreenText="Video 2 minutes 15 seconds" />` |
| topStory | boolean | No      | false   | true          |

## Headline Props

<!-- prettier-ignore -->
| Argument  | Type | Required | Default | Example |
| --------- | ---- | -------- | ------- | ------- |
| Script    | object | yes | latin | { canon: { groupA: { fontSize: '28', lineHeight: '32',}, groupB: { fontSize: '32', lineHeight: '36', }, groupD: { fontSize: '44', lineHeight: '48', }, }, trafalgar: { groupA: { fontSize: '20', lineHeight: '24', }, groupB: { fontSize: '24', lineHeight: '28', }, groupD: { fontSize: '32', lineHeight: '36', }, }, }|
| service | string | yes | N/A | `'news'` |

## Summary Props

<!-- prettier-ignore -->
| Argument  | Type | Required | Default | Example |
| --------- | ---- | -------- | ------- | ------- |
| Script    | object | yes | latin | { canon: { groupA: { fontSize: '28', lineHeight: '32',}, groupB: { fontSize: '32', lineHeight: '36', }, groupD: { fontSize: '44', lineHeight: '48', }, }, trafalgar: { groupA: { fontSize: '20', lineHeight: '24', }, groupB: { fontSize: '24', lineHeight: '28', }, groupD: { fontSize: '32', lineHeight: '36', }, }, }|
| service | string | yes | N/A | `'news'` |

## Usage

The typical use-case of this component is as displayed below. A Image sits on the left side of the promo with info elements on the right. These info elements are typically a headline, text summary paragraph and timestamp. The `Headline` and `Summary` components are provided by this package and can be imported as seen below.

This component also has an option to display a media indicator, which consists of a play icon and duration of the media, if that data is provided.

The `topStory` prop can be passed to adopt a vertical card layout under 600px. This is designed to be used only on the first (top) story on an index. This prop must be passed to the StoryPromo, Headline and Summary components.

```jsx
import React, { Fragment } from 'react';
import StoryPromo, { Headline, Summary, Link } from '@bbc/psammead-story-promo';
import { latin } from '@bbc/gel-foundations/scripts';

const Image = <img src="https://foobar.com/image.jpg" />;

const Info = (
  <Fragment>
    <Headline script={latin} topStory={true}>
      <Link href="https://www.bbc.co.uk/news">The headline of the promo</Link>
    </Headline>
    <Summary script={latin} topStory={true}>
      The summary of the promo
    </Summary>
    <time>12 March 2019</time>
  </Fragment>
);

<StoryPromo image={Image} info={Info} topStory={true} />;
```

### When to use this component

The `StoryPromo` component is designed to be used within a link element to allow the user to navigate to the story on another page.

<!-- ### When not to use this component -->

### Accessibility notes

This component uses full semantic markup for the `Headline`, `Summary`, and `Link`, using `h3`, `p` and `a` respectively. Other accessibility factors such as image alt text and time elements are passed in as props and aren't explicitly set in this component.

The link is nested inside the `h3` for better support with VoiceOver Mac and Safari. We use the `faux block link` pattern which makes the entire block clickable, whilst also enabling links nested within in that block to also be clickable.

We haven't yet thoroughly looked at cross device browser AT testing yet, this is in-progress. We will update with a11y notes when carrying out a proper release.

## Roadmap

This component will soon be integrated on to add support for additional UX and product requirements such as;

- Hover and focus states
- Logic to toggle a 'Top Story' mode which has greater text size and a vertically stacked design

## Contributing

Psammead is completely open source. We are grateful for any contributions, whether they be new components, bug fixes or general improvements. Please see our primary contributing guide which can be found at [the root of the Psammead repository](https://github.com/bbc/psammead/blob/latest/CONTRIBUTING.md).

### [Code of Conduct](https://github.com/bbc/psammead/blob/latest/CODE_OF_CONDUCT.md)

We welcome feedback and help on this work. By participating in this project, you agree to abide by the [code of conduct](https://github.com/bbc/psammead/blob/latest/CODE_OF_CONDUCT.md). Please take a moment to read it.

### License

Psammead is [Apache 2.0 licensed](https://github.com/bbc/psammead/blob/latest/LICENSE).
