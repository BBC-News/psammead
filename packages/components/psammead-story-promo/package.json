{
  "name": "@bbc/psammead-story-promo",
  "version": "1.0.0-alpha.3",
  "main": "dist/index.js",
  "description": "A story promo for use on index pages",
  "repository": {
    "type": "git",
    "url": "https://github.com/bbc/psammead/tree/latest/packages/components/psammead-story-promo"
  },
  "author": {
    "name": "Psammead Maintainers",
    "email": "PsammeadMaintainers@bbc.co.uk"
  },
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/bbc/psammead/issues"
  },
  "homepage": "https://github.com/bbc/psammead/blob/latest/packages/components/psammead-story-promo/README.md",
  "dependencies": {
    "@bbc/gel-foundations": "^3.0.0",
    "@bbc/psammead-styles": "^0.3.3"
  },
  "devDependencies": {
    "@bbc/psammead-image": "^0.3.5",
    "@bbc/psammead-media-indicator": "^0.1.0",
    "@bbc/psammead-storybook-helpers": "^2.0.0",
    "@bbc/psammead-test-helpers": "^0.3.3",
    "@bbc/psammead-timestamp": "^0.3.2",
    "react": "^16.8.6",
    "styled-components": "^4.3.2"
  },
  "peerDependencies": {
    "prop-types": "^15.7.2",
    "react": "^16.8.6",
    "styled-components": "^4.3.2"
  },
  "keywords": [
    "bbc",
    "story",
    "promo"
  ],
  "publishConfig": {
    "tag": "alpha"
  }
}
