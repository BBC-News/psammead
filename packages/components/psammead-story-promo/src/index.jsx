import React from 'react';
import styled, { css } from 'styled-components';
import { node, bool } from 'prop-types';
import {
  GEL_SPACING,
  GEL_SPACING_DBL,
  GEL_GUTTER_BELOW_600PX,
  GEL_GUTTER_ABOVE_600PX,
} from '@bbc/gel-foundations/spacings';
import {
  GEL_GROUP_2_SCREEN_WIDTH_MIN,
  GEL_GROUP_2_SCREEN_WIDTH_MAX,
  GEL_GROUP_3_SCREEN_WIDTH_MIN,
  GEL_GROUP_5_SCREEN_WIDTH_MIN,
} from '@bbc/gel-foundations/breakpoints';
import {
  getPica,
  getGreatPrimer,
  getParagon,
  getLongPrimer,
} from '@bbc/gel-foundations/typography';
import { C_EBON, C_SHADOW, C_METAL } from '@bbc/psammead-styles/colours';
import { getSansRegular, getSerifBold } from '@bbc/psammead-styles/font-styles';

const twoOfSixColumnsMaxWidthScaleable = `33.33%`;
// (2 / 6) * 100 = 0.3333333333 = 33.33%

const fourOfTwelveColumnsMaxWidthScaleable = `33.33%`;
// (4 / 12) * 100 = 0.3333333333 = 33.33%

const fourOfSixColumnsMaxWidthScaleable = `66.67%`;
// (4 / 6) * 100 = 66.6666666667 = 66.67%

const eightOfTwelveColumnsMaxScaleable = `66.67%`;
// (8 / 12) * 100 = 66.6666666667 = 66.67%

const fullWidthColumnsMaxScaleable = `100%`;
// (12 / 12) * 100 = 100 = 100%

const StoryPromoWrapper = styled.div`
  position: relative;

  @supports (display: grid) {
    display: grid;
    grid-template-columns: repeat(6, 1fr);
    grid-column-gap: ${GEL_GUTTER_BELOW_600PX};

    @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
      grid-column-gap: ${GEL_GUTTER_ABOVE_600PX};
    }

    @media (min-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN}) {
      grid-template-columns: repeat(12, 1fr);
    }
  }
`;

const ImageGridColumnsTopStory = css`
  grid-column: 1 / span 6;

  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    grid-column: 1 / span 2;
  }

  @media (min-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN}) {
    grid-column: 1 / span 4;
  }
`;

const ImageGridColumns = css`
  grid-column: 1 / span 2;

  @media (min-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN}) {
    grid-column: 1 / span 4;
  }
`;

const ImageGridFallbackTopStory = css`
  margin-bottom: ${GEL_GUTTER_BELOW_600PX};
  width: ${fullWidthColumnsMaxScaleable};

  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    width: ${twoOfSixColumnsMaxWidthScaleable};
    margin-bottom: none;
  }
`;

const ImageGridFallback = css`
  width: ${twoOfSixColumnsMaxWidthScaleable};
`;

const ImageGridItem = styled.div`
  display: inline-block;
  vertical-align: top;
  position: relative;

  ${({ topStory }) =>
    topStory ? ImageGridFallbackTopStory : ImageGridFallback}

  @media (min-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN}) {
    width: ${fourOfTwelveColumnsMaxWidthScaleable};
  }

  @supports (display: grid) {
    display: block;
    width: initial;

    ${({ topStory }) =>
      topStory ? ImageGridColumnsTopStory : ImageGridColumns}
  }
`;

const ImageContentsWrapper = styled.div`
  position: relative;
`;

const InlineMediaIndicator = styled.div`
  ${({ topStory }) =>
    topStory
      ? `
      position: absolute;
      bottom: 0;
      `
      : `
      @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
        position: absolute;
        bottom: 0;
      }
      `}
`;

export const Headline = styled.h3`
  ${props => (props.script ? getPica(props.script) : '')};

  ${({ script, topStory }) => {
    if (!script) {
      return '';
    }

    return topStory ? getParagon(script) : getPica(script);
  }};

  color: ${C_EBON};
  ${({ service }) => getSerifBold(service)}
  margin: 0; /* Reset */
  padding-bottom: ${GEL_SPACING};
  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    ${props => (props.script ? getGreatPrimer(props.script) : '')};
  }
`;

export const Summary = styled.p`
  ${props => (props.script ? getLongPrimer(props.script) : '')};
  color: ${C_SHADOW};
  ${({ service }) => getSansRegular(service)}
  margin: 0; /* Reset */
  padding-bottom: ${GEL_SPACING};

  ${({ topStory }) =>
    !topStory &&
    css`
      @media (max-width: ${GEL_GROUP_2_SCREEN_WIDTH_MAX}) {
        display: none;
        visibility: hidden;
      }
    `}
`;

const TextGridColumnsTopStory = css`
  grid-column: 1 / span 6;

  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    grid-column: 3 / span 4;
  }

  @media (min-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN}) {
    grid-column: 5 / span 8;
  }
`;

const TextGridColumns = css`
  grid-column: 3 / span 4;

  @media (min-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN}) {
    grid-column: 5 / span 8;
  }
`;

const TextGridFallbackTopStory = css`
  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    width: ${fourOfSixColumnsMaxWidthScaleable};
  }
`;

const TextGridFallback = css`
  width: ${fourOfSixColumnsMaxWidthScaleable};
  padding: 0 ${GEL_SPACING};
`;

const TextGridItem = styled.div`
  display: inline-block;
  vertical-align: top;

  ${({ topStory }) => (topStory ? TextGridFallbackTopStory : TextGridFallback)}

  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    padding: 0 ${GEL_SPACING_DBL};
  }

  @media (min-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN}) {
    width: ${eightOfTwelveColumnsMaxScaleable};
  }

  @supports (display: grid) {
    display: block;
    width: initial;
    padding: initial;

    ${({ topStory }) => (topStory ? TextGridColumnsTopStory : TextGridColumns)}
  }
`;

export const Link = styled.a`
  position: static;
  color: ${C_EBON};
  text-decoration: none;

  &:before {
    bottom: 0;
    content: '';
    left: 0;
    overflow: hidden;
    position: absolute;
    right: 0;
    top: 0;
    white-space: nowrap;
    z-index: 1;
  }

  &:hover,
  &:focus {
    text-decoration: underline;
  }

  &:visited {
    color: ${C_METAL};
  }
`;

const StoryPromo = ({ image, info, mediaIndicator, topStory }) => (
  <StoryPromoWrapper>
    <ImageGridItem topStory={topStory}>
      <ImageContentsWrapper>
        {image}
        {mediaIndicator && (
          <InlineMediaIndicator topStory={topStory}>
            {mediaIndicator}
          </InlineMediaIndicator>
        )}
      </ImageContentsWrapper>
    </ImageGridItem>
    <TextGridItem topStory={topStory}>{info}</TextGridItem>
  </StoryPromoWrapper>
);

StoryPromo.propTypes = {
  image: node.isRequired,
  info: node.isRequired,
  mediaIndicator: node,
  topStory: bool,
};

StoryPromo.defaultProps = {
  mediaIndicator: null,
  topStory: false,
};

export default StoryPromo;
