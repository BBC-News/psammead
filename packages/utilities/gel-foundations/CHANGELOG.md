# Gel Foundations Changelog

| Version | Description |
|---------|-------------|
| 0.1.4   | [PR#224](https://github.com/BBC-News/psammead/pull/224) Add tests for the exported values, coverage 100% :tada: |
| 0.1.1   | [PR#212](https://github.com/BBC-News/psammead/pull/212) Update package description and README. |
| 0.1.0   | [PR#221](https://github.com/BBC-News/psammead/pull/221) Create initial package, pulled in from gel-constants and gel-foundations-styled-components packages. |

