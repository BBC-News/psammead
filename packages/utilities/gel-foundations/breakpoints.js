module.exports = require('./dist/breakpoints'); // eslint-disable-line import/no-unresolved
