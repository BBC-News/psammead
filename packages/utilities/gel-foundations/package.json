{
  "name": "drew-testing-123",
  "version": "1.0.6",
  "main": "dist/index.cjs.js",
  "module": "dist/index.es.js",
  "description": "A range of string constants for use in CSS, intended to help implement BBC GEL-compliant webpages and components.",
  "repository": {
    "type": "git",
    "url": "https://github.com/bbc/psammead/tree/latest/packages/utilities/gel-foundations"
  },
  "author": {
    "name": "Psammead Maintainers",
    "email": "PsammeadMaintainers@bbc.co.uk"
  },
  "files": [
    "dist/"
  ],
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/bbc/psammead/issues"
  },
  "homepage": "https://github.com/bbc/psammead/blob/latest/packages/utilities/gel-foundations/README.md",
  "keywords": [
    "bbc",
    "utilities",
    "gel",
    "breakpoints",
    "grid",
    "spacing",
    "typography"
  ],
  "devDependencies": {
    "@bbc/psammead-test-helpers": "^1.0.1"
  }
}
