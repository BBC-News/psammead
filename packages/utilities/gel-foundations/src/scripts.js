export latin from './scripts/latin-and-cyrillic';
export cyrillic from './scripts/latin-and-cyrillic';
export latinDiacritics from './scripts/latin-with-diacritics';
export arabic from './scripts/arabic';
export arabicPashto from './scripts/arabic-pashto';
export bengali from './scripts/bengali';
export burmese from './scripts/burmese';
export hindi from './scripts/devanagari-and-gurmukhi';
export nepali from './scripts/devanagari-and-gurmukhi';
export sinhalese from './scripts/sinhalese';
export tamil from './scripts/tamil';
export thai from './scripts/thai';
export chinese from './scripts/no-ascenders-or-descenders';
export korean from './scripts/no-ascenders-or-descenders';
export ethiopic from './scripts/no-ascenders-or-descenders';
