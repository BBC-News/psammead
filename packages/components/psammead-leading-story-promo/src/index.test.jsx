import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { latin, arabic } from '@bbc/gel-foundations/scripts';
import { Summary, Link } from '@bbc/psammead-story-promo';
import LeadingStoryPromo, { LeadingPromoHeadline } from './index';

const Image = <img src="https://foobar.com/image.png" alt="Alt text" />;

// eslint-disable-next-line react/prop-types
const Info = (script, headlineText, summaryText) => (
  <>
    <LeadingPromoHeadline script={script} service="news">
      <Link href="https://www.bbc.co.uk/news">{headlineText}</Link>
    </LeadingPromoHeadline>
    <Summary script={script} service="news">
      {summaryText}
    </Summary>
  </>
);

describe('LeadingStoryPromo', () => {
  shouldMatchSnapshot(
    'should render default leading story promo correctly',
    <LeadingStoryPromo
      image={Image}
      info={Info(
        latin,
        'The headline of the promo',
        'The summary of the promo',
      )}
    />,
  );

  shouldMatchSnapshot(
    'should render RTL leading story promo correctly',
    <LeadingStoryPromo
      image={Image}
      info={Info(arabic, 'هذا هو العنوان الخاص بي', 'هذا عنوان فرعي')}
      dir="rtl"
    />,
  );
});
