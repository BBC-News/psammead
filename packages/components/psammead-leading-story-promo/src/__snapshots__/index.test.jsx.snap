// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeadingStoryPromo should render RTL leading story promo correctly 1`] = `
.c4 {
  font-size: 1.0625rem;
  line-height: 1.5rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #3F3F42;
  margin: 0;
  padding-bottom: 0.5rem;
}

.c3 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c3:hover,
.c3:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3:visited {
  color: #6E6E73;
}

.c5 {
  display: inline-block;
  width: 66.67%;
}

.c1 {
  display: inline-block;
  vertical-align: top;
  width: 33.33%;
}

.c2 {
  font-size: 1.5rem;
  line-height: 2rem;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  color: #222222;
  margin: 0;
  padding-bottom: 0.5rem;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 1.125rem;
    line-height: 1.5rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1.125rem;
    line-height: 1.5rem;
  }
}

@media (max-width:37.4375rem) {
  .c4 {
    display: none;
    visibility: hidden;
  }
}

@media (min-width:63rem) {
  .c4 {
    display: none;
    visibility: hidden;
  }
}

@supports (grid-template-columns:fit-content(200px)) {
  .c0 {
    display: grid;
    grid-template-columns: repeat(6,1fr);
    grid-column-gap: 1rem;
  }
}

@supports (grid-template-columns:fit-content(200px)) {
  .c5 {
    width: initial;
  }
}

@supports (grid-template-columns:fit-content(200px)) {
  .c1 {
    width: 100%;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c2 {
    font-size: 1.625rem;
    line-height: 2.25rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    font-size: 1.75rem;
    line-height: 2.25rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h3
      class="c2"
    >
      <a
        class="c3"
        href="https://www.bbc.co.uk/news"
      >
        هذا هو العنوان الخاص بي
      </a>
    </h3>
    <p
      class="c4"
    >
      هذا عنوان فرعي
    </p>
  </div>
  <div
    class="c5"
  >
    <img
      alt="Alt text"
      src="https://foobar.com/image.png"
    />
  </div>
</div>
`;

exports[`LeadingStoryPromo should render default leading story promo correctly 1`] = `
.c4 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #3F3F42;
  margin: 0;
  padding-bottom: 0.5rem;
}

.c3 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c3:hover,
.c3:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3:visited {
  color: #6E6E73;
}

.c5 {
  display: inline-block;
  width: 66.67%;
}

.c1 {
  display: inline-block;
  vertical-align: top;
  width: 33.33%;
}

.c2 {
  font-size: 1.125rem;
  line-height: 1.375rem;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  color: #222222;
  margin: 0;
  padding-bottom: 0.5rem;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (max-width:37.4375rem) {
  .c4 {
    display: none;
    visibility: hidden;
  }
}

@media (min-width:63rem) {
  .c4 {
    display: none;
    visibility: hidden;
  }
}

@supports (grid-template-columns:fit-content(200px)) {
  .c0 {
    display: grid;
    grid-template-columns: repeat(6,1fr);
    grid-column-gap: 1rem;
  }
}

@supports (grid-template-columns:fit-content(200px)) {
  .c5 {
    width: initial;
  }
}

@supports (grid-template-columns:fit-content(200px)) {
  .c1 {
    width: 100%;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c2 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    font-size: 1.5rem;
    line-height: 1.75rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h3
      class="c2"
    >
      <a
        class="c3"
        href="https://www.bbc.co.uk/news"
      >
        The headline of the promo
      </a>
    </h3>
    <p
      class="c4"
    >
      The summary of the promo
    </p>
  </div>
  <div
    class="c5"
  >
    <img
      alt="Alt text"
      src="https://foobar.com/image.png"
    />
  </div>
</div>
`;
